libkbfs/folder_block_ops.go

```diff
@@ -396,6 +397,75 @@ func (fbo *folderBlockOps) GetBlockForReading(ctx context.Context,
 		NewCommonBlock, false, path{}, blockRead)
 }
 
+// GetBlocksForReading retrieves the blocks pointed to by ptrs, all of
+// which must be valid, either from the cache or from the server.  The
+// returned block may have a generic type (not DirBlock or FileBlock).
+//
+// The caller can specify a set of pointers using
+// `ignoreRecoverableErrors` for which "recoverable" fetch errors are
+// tolerated.  In that case, the returned map will not have an entry
+// for any pointers in the `ignoreRecoverableErrors` set that hit such
+// an error.
+//
+// This should be called for "internal" operations, like conflict
+// resolution and state checking, which don't know what kind of block
+// the pointers refer to.  The blocks will not be cached, if they
+// weren't in the cache already.
+func (fbo *folderBlockOps) GetBlocksForReading(ctx context.Context,
+	lState *lockState, kmd KeyMetadata, ptrs []BlockPointer,
+	ignoreRecoverableErrors map[BlockPointer]bool,
+	branch BranchName) (map[BlockPointer]Block, error) {
+	fbo.blockLock.RLock(lState)
+	defer fbo.blockLock.RUnlock(lState)
+
+	type resp struct {
+		ptr   BlockPointer
+		block Block
+	}
+
+	respChans := make([]<-chan resp, 0, len(ptrs))
+	eg, groupCtx := errgroup.WithContext(ctx)
+	for _, ptr := range ptrs {
+		ptr := ptr
```

Comment:
I understand what you're doing here, but I don't love shadowing names because of the ambiguity it creates. Can you use a different name for the iteration variable?